app/assets/javascripts/modules/GeoLocate.js

```diff
@@ -0,0 +1,71 @@
+define(['jquery'], function($) {
+
+  'use strict';
+
+  var location = $('[data-geo-location]'),
+      description = $('.search-filter__description'),
+      input = $('[data-geo-location-input]'),
+      label = $('[data-geo-location-label]'),
+      container = $('[data-geo-location-container]'),
+      button = $('[data-geo-location-button]'),
+      geocoder,
+      postcode;
+
+  function initialise() {
+    geocoder = new google.maps.Geocoder();
+  }
+
+  if (navigator.geolocation) {
+    navigator.geolocation.getCurrentPosition(successFunction, errorFunction);
```

Comment:
I know this is still WIP, but you might want to consider grabbing the location when the user asks for it by clicking on a button or something, rather than just when this module is pulled through. Can be less jarring, and reduces the risk of people missing the 'whats your location' browser prompt.